src/theme.c

```diff
@@ -1042,3 +1044,40 @@ theme_finish(struct theme *theme)
 	zdrop(&theme->corner_top_right_active_normal);
 	zdrop(&theme->corner_top_right_inactive_normal);
 }
+
+struct theme
+get_theme_for_view(struct view *view)
+{
+	float *custom_color = window_rules_get_custom_border_color(view);
+	if (!custom_color) {
+		return *view->server->theme;
+	}
+
+	struct theme theme = { 0 };
+
+	/* The same code as in theme_init*(), but with introducing custom color*/
+	theme_builtin(&theme);
+
+	struct wl_list paths;
+	paths_theme_create(&paths, rc.theme_name, "themerc");
+	theme_read(&theme, &paths);
```

Comment:
Oh, reading the theme multiple times doesn't feel right. Could we make copies of it (on the heap)? Is it the creation of the rounded corners that becomes a problem?